src/internal.c

```diff
@@ -28823,20 +28928,46 @@ int SendClientKeyExchange(WOLFSSL* ssl)
                 case psk_kea:
                 {
                     byte* pms = ssl->arrays->preMasterSecret;
+                #ifdef WOLFSSL_MAXQ10XX_TLS
+                    byte dummy_psk[MAX_PSK_KEY_LEN];
+                    word32 ret_cb;
+                    /* get client_identity */
+                    ret_cb = ssl->options.client_psk_cb(ssl,
+                                 ssl->arrays->server_hint,
+                                 ssl->arrays->client_identity,
+                                 MAX_PSK_ID_LEN, dummy_psk,
+                                 MAX_PSK_KEY_LEN);
+                    if (ret_cb == 0 || ret_cb > MAX_PSK_KEY_LEN) {
```

Comment:
Does ret_cb have the length? If so, I think you should be putting the \0 at line 28945 just after the length (i.e. after the first ret_cb bytes), not at the end of the buffer.

Reply:
This code has been removed and replaced with generic return  code logic.  This is in 7d04d275f